### PR TITLE
jobs: add resource_class parameters to all jobs

### PR DESCRIPTION
**Pull request by kopiczko**
While switching to architect orb within Go modules work we discovered
that some codebases require more resources for particular jobs. The
problem we hit was aws-operator requiring more memory for building the
binary.

This change adds `resource_class` parameter to all (excluding legacy)
jobs. If not specified in CircleCI the default value is always "medium"
for both machine and docker executors. For some jobs the default is
changed to "small" to save some money.

CircleCI pricing table: https://circleci.com/pricing/#compute-options-table

- [x] src/jobs/changelog-lint.yaml - small by default.
- [x] src/jobs/gitleaks.yaml - small by default.
- [x] src/jobs/go-build.yaml - medium by default.
- [x] src/jobs/go-test.yaml - medium by default.
- [x] src/jobs/integration-test.yaml - medium by default.
- [x] src/jobs/push-to-app-catalog.yaml - small by default.
- [x] src/jobs/push-to-app-collection.yaml - small by default.
- [x] src/jobs/push-to-docker.yaml - small by default.
- [x] src/jobs/go-architect-legacy.yaml **SKIP** legacy job.
- [x] src/jobs/go-test-legacy.yaml **SKIP** legacy job.
- [x] src/jobs/push-to-docker-legacy.yaml **SKIP** legacy job.

## Checklist

- [x] Make yourself familiar with following readme sections:
    - [x] [Coding Standards](https://github.com/giantswarm/architect-orb#coding-guidelines).
    - [x] [Development](https://github.com/giantswarm/architect-orb#development).
    - [x] [Design and Goals](https://github.com/giantswarm/architect-orb#design-and-goals).
- [x] :warning: Update changelog in [CHANGELOG.md](https://github.com/giantswarm/architect-orb/tree/master/CHANGELOG.md).
- [x] :warning: After the release update architect orb version in [.circleci/config.yml](https://github.com/giantswarm/architect-orb/tree/master/.circleci/config.yml).

### CODE REVIEW

File: docs/integration_test_job.md [hunk 1]
```diff
@@ -25,7 +23,14 @@ workflows:
...
 
-These optional parameters allow configuring the test.
+- [common parameters](common.md#parameters) shared in all jobs.
+- [test-dir](#attach_workspace) (required string)
+- [env-file](#env-file) (optional string, default="")
+- [kind-config](#kind-config) (optional string, default="")
+- [kubernetes-version](#kubernetes-version) (optional string, default="v1.16.3")
+- [setup-script](#setup-script) (optional string, default="")
+- [test-dir](#test-dir) (required string)
+- [test-timeout](#test-timeout) (optional string, default="20m")
```
Comment (kopiczko) [inline docs/integration_test_job.md:33]: @rossf7 I changed the structure of the docs a bit. Let me know if you don't like it :D

**Comment by rossf7** [comment]
OK nice if we're increasing integration test from small to medium this may help with the failing KIND clusters.

Circle support suggested increasing the resource class.

https://github.com/giantswarm/giantswarm/issues/9520#issuecomment-611472617

File: docs/integration_test_job.md (hunk 1 above)
Comment (rossf7) [reply docs/integration_test_job.md:33]: Fine with me. Thx!

**Comment by kopiczko** [comment]
@rossf7 medium is the default so nothing has changed. Feel free to change that here or you can experiment in your job definition by specifying resource_class there.

**Pull request merged**